cmd/kubeadm/app/cmd/join.go

```diff
@@ -413,87 +412,10 @@ func (j *joinData) OutputWriter() io.Writer {
 	return j.outputWriter
 }
 
-// Run executes worker node provisioning and tries to join an existing cluster.
-func (j *joinData) Run() error {
-	// Fetch the init configuration based on the join configuration.
-	// TODO: individual phases should call these:
-	//   - phases that need initCfg should call joinData.InitCfg().
-	//   - phases that need tlsBootstrapCfg should call joinData.TLSBootstrapCfg().
-	initCfg, err := j.InitCfg()
-	if err != nil {
-		return err
-	}
-
-	// if the node is hosting a new control plane instance
-	if j.cfg.ControlPlane != nil {
```

Comment:
If I'm not wrong this condition is not reflected in the new phases (the `etcd-local`, `upload-config`, `mark-control-plane` phases should be executed only if `j.cfg.ControlPlane != nil`; see e.g what already implemented for `control-plane-prepare`)